### PR TITLE
bt: esp32: add hal support for bluetooth

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -74,9 +85,55 @@ if(CONFIG_SOC_ESP32)
...
+  ## shared WIFI/BT resources
+  if (CONFIG_BT OR CONFIG_WIFI_ESP32)
+    zephyr_sources(
+      ../components/esp_wifi/src/phy_init.c
+      adapter/src/common/esp_system_api.c
+      )
+
+    zephyr_link_libraries(
+      net80211
+      core
```
Comment (uLipe) [inline zephyr/CMakeLists.txt:101]: Question: the wifi related libraries are all needed even when using BT only?

File: zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -74,9 +85,55 @@ if(CONFIG_SOC_ESP32)
...
+        -L${CMAKE_CURRENT_SOURCE_DIR}/../components/esp32
+        -L${CMAKE_CURRENT_SOURCE_DIR}/../components/esp_wifi/lib/esp32
+        -Wl,--defsym,phy_printf=printf
+        -Wl,--defsym,net80211_printf=printf
+        -Wl,--defsym,coexist_printf=printf
+      )
+
+  endif()
+
+  ## BT definitions
```
Comment (uLipe) [inline zephyr/CMakeLists.txt:117]: I think this comment is unnecessary since the configuration option is self explanatory.

**Review by uLipe — COMMENTED** [review]
Two minor questions, otherwise LGTM.

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (sylvioalves) [reply zephyr/CMakeLists.txt:101]: yes, it is necessary as it shares phy and radio stuff.

**Review by uLipe — APPROVED** [review]
Okay, LGTM.

**Review by shubhamkulkarni97 — APPROVED** [review]

**Review by glaubermaroto — APPROVED** [review]

**Pull request merged**